src/test/java/edu/harvard/iq/dataverse/util/cache/RateLimitUtilTest.java

```diff
@@ -0,0 +1,128 @@
+package edu.harvard.iq.dataverse.util.cache;
+
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.authorization.users.GuestUser;
+import edu.harvard.iq.dataverse.authorization.users.User;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import edu.harvard.iq.dataverse.util.cache.RateLimitUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class RateLimitUtilTest {
+
+    private SystemConfig mockedSystemConfig;
+
+    static final String settingJson = "{\n" +
+            "  \"rateLimits\":[\n" +
+            "    {\n" +
+            "      \"tier\": 0,\n" +
+            "      \"limitPerHour\": 10,\n" +
+            "      \"actions\": [\n" +
+            "        \"GetLatestPublishedDatasetVersionCommand\",\n" +
+            "        \"GetPrivateUrlCommand\",\n" +
+            "        \"GetDatasetCommand\",\n" +
+            "        \"GetLatestAccessibleDatasetVersionCommand\"\n" +
+            "      ]\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"tier\": 0,\n" +
+            "      \"limitPerHour\": 1,\n" +
+            "      \"actions\": [\n" +
+            "        \"CreateGuestbookResponseCommand\",\n" +
+            "        \"UpdateDatasetVersionCommand\",\n" +
+            "        \"DestroyDatasetCommand\",\n" +
+            "        \"DeleteDataFileCommand\",\n" +
+            "        \"FinalizeDatasetPublicationCommand\",\n" +
+            "        \"PublishDatasetCommand\"\n" +
+            "      ]\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"tier\": 1,\n" +
+            "      \"limitPerHour\": 30,\n" +
+            "      \"actions\": [\n" +
+            "        \"CreateGuestbookResponseCommand\",\n" +
+            "        \"GetLatestPublishedDatasetVersionCommand\",\n" +
+            "        \"GetPrivateUrlCommand\",\n" +
+            "        \"GetDatasetCommand\",\n" +
+            "        \"GetLatestAccessibleDatasetVersionCommand\",\n" +
+            "        \"UpdateDatasetVersionCommand\",\n" +
+            "        \"DestroyDatasetCommand\",\n" +
+            "        \"DeleteDataFileCommand\",\n" +
+            "        \"FinalizeDatasetPublicationCommand\",\n" +
+            "        \"PublishDatasetCommand\"\n" +
+            "      ]\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}";
+    static final String settingJsonBad = "{\n";
+
+    @BeforeEach
+    public void setup() {
```

Comment:
**suggestion**: It might make sense to use inner test classes to avoid setting these things before each test. Might be bikesheeding, as we only have a very limited number of tests in this class anyway.

Reply:
I rearranged the code but it still needs to clear the list and the map before each test. I don't think nested tests would improve the tests